### DISCUSSION TITLE
Hardware platforms

### DISCUSSION THREAD

**Discussion by otalado** [dail8859/NotepadNext#251] [Ideas]
I just tumbled to NotepadNext and was pleasantly surprised to see how quick it is. I like the fact that it is not a resources hog (like some other editors). So I have a quick question. Is there a release for ARM64 anywhere in the future of NotepadNext? It'd be nice to have an editor like this on RPI4 ...

**Comment by dail8859** [comment]
I personally don't have any plans to add arm64 support myself since I don't use it or know much about it.

However, if someone is interested in adding support for arm64 and it doesn't add too much burden or overhead to building/maintaining the application then I'd be glad to have the capabilities. I'd assume Qt would run fine on RPI4 since it is such a widespread framework and targeted to various platforms.

**Reply by ZachBacon** [reply]
Considering KDE plasma can operate on arm, it can operate. In fact I even managed to get notepadnext  working on windows arm64. No modifications necessary, so the poster of this issue could compile it themselves

**Comment by otalado** [comment]
Well, I did not know how to go about building NotepadNext from the sources. Then I (finally)  tripped over the Building.md in the docs. So I gave the Linux recipe a try. It chugged a few minutes and the NotepadNext was successfully built. I am ashamed to admit I don't have a clue how this make process works (but it works). I have a working NotepadNext on my RPI4 now. So far I haven't noticed any problems with it. Now I'll try to figure out how to create .appimage for the editor.

**Reply by dail8859** [reply]
Just had a thought. Are you able to install the flatpak distribution of Notepad Next?

According to their documentation it supports RPI: https://flatpak.org/setup/Raspberry%20Pi%20OS

```
flatpak install flathub com.github.dail8859.NotepadNext
```

**Reply by otalado** [reply]
Ok, I will try that and report back to you what has happened sometime this weekend.

**Reply by otalado** [reply]
So I first tried the flatpak distro on RPI4 Buster (32bit). It did not work. I read somewhere the flatpack is phasing out 32 bit support so I created Bullseye (64bit) disk and booted from there - with the same result:
```
pi@pi4-4:~ $ sudo flatpak install flathub com.github.dail8859.NotepadNext
Looking for matches…
error: No remote refs found similar to ‘flathub’
pi@pi4-4:~ $
```
The error message leaves me to think of two possible reasons: 
a) the install command is not correct(?) or 
b) I installed flatpak wrong, i.e. **apt install flatpak**  command is not enough, perhaps additional steps are needed (?) 

As I have zero experience with flatpak, I don't even know where/how to start troubleshooting. 

If you have any ideas about what I can try, I am willing to try it because I like this editor and it'll be nice to have an install package for it for the linux ARM devices..

**Reply by otalado** [reply]
It turns out that an additional step for the flatpak was needed:
`sudo flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo`
After that the command:
`sudo flatpak install flathub com.github.dail8859.NotepadNext` worked. 
The NotepadNext was installed in a kilometer-long path under **/var**. I had to change the privileges. When I run the NotepadNext I got the following errors:
```
pi@bullseye:/var/lib/flatpak/app/com.github.dail8859.NotepadNext/aarch64/stable/21d7b27e0b42bbe032037427b85d1f2dc1a786c97246972cbc2f59ab1c70dfe2/files/bin $ ./NotepadNext
./NotepadNext: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./NotepadNext)
./NotepadNext: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./NotepadNext)
./NotepadNext: /lib/aarch64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by ./NotepadNext)
./NotepadNext: /lib/aarch64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by ./NotepadNext)
pi@bullseye:
```
I am afraid that this is a show-stopper, messing with libc is never a good idea. On a fresh Raspberry Pi Os (Bullseye) the instlled libc is version 2.31, so ...

**Reply by dail8859** [reply]
It was worth a shot at least. Thanks for trying. I personally don't know too much about the flatpak distribution.

@zocker-160 Any thoughts on this?

**Comment by zocker-160** [comment]
Please use "flatpak run" for launching the application as specified in the instructions provided by Flathub.org. No changing of privileges or messing with libc is necessary.

It works perfectly on my Pi4, so it should on yours too, just keep in mind, that only 64bit ARM is supported.

**Reply by otalado** [reply]
I tried to use **flatpack run** and ended up with complaint:

```
pi@bullseye: $ flatpak run NotepadNext

Note that the directories 

'/var/lib/flatpak/exports/share'
'/home/pi/.local/share/flatpak/exports/share'

are not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

error: Invalid id NotepadNext: Names must contain at least 2 periods
```
I then went to investigate what is in those two folders. In **/var/lib/flatpak/exports/share/applications**  folder I found **Notepad Next.desktop** file. When I double click on it the NotepadNext starts and works like expected.

So now I'll go to flapak docs and try to figure out how to properly set it up, so that it will install NotepadNext without such a mess. Or, perhaps @zocker-160 can give brief instructions on that?

**Comment by zocker-160** [comment]
All you need to do is follow the official instructions carefully step by step and then it will work.

Installation of Flatpak on a RPI:

https://flatpak.org/setup/Raspberry%20Pi%20OS


Installation and running of Notepad Next (scroll down):

https://flathub.org/apps/details/com.github.dail8859.NotepadNext


I order to update all Flatpaks you simply run "flatpak update".

I sadly cannot help you more than this, because my phone is everything I have available to me currently. :(

>Am 30. Oktober 2022 08:02:53 MEZ schrieb otalado ***@***.***>:
>I tried to use **flatpack run** and ended up with complaint:
>
>```
>***@***.***: $ flatpak run NotepadNext
>
>Note that the directories 
>
>'/var/lib/flatpak/exports/share'
>'/home/pi/.local/share/flatpak/exports/share'
>
>are not in the search path set by the XDG_DATA_DIRS environment variable, so
>applications installed by Flatpak may not appear on your desktop until the
>session is restarted.
>
>error: Invalid id NotepadNext: Names must contain at least 2 periods
>```
>I then went to investigate what is in those two folders. In **/var/lib/flatpak/exports/share/applications**  folder I found **Notepad Next.desktop** file. When I double click on it the NotepadNext starts and works like expected.
>
>So now I'll go to flapak docs and try to figure out how to properly set it up, so that it will install NotepadNext without such a mess. Or, perhaps @zocker-160 can give brief instructions on that?
>
>
